src/app/services/task.service.ts

```diff
@@ -14,102 +14,102 @@ export class TaskService {
-    { label: '08:30:00', value: '08:30:00' },
-    { label: '08:45:00', value: '08:45:00' },
-    { label: '09:00:00', value: '09:00:00' },
-    { label: '09:15:00', value: '09:15:00' },
-    { label: '09:30:00', value: '09:30:00' },
-    { label: '09:45:00', value: '09:45:00' },
-    { label: '10:00:00', value: '10:00:00' },
-    { label: '10:15:00', value: '10:15:00' },
-    { label: '10:30:00', value: '10:30:00' },
-    { label: '10:45:00', value: '10:45:00' },
-    { label: '11:00:00', value: '11:00:00' },
-    { label: '11:15:00', value: '11:15:00' },
-    { label: '11:30:00', value: '11:30:00' },
-    { label: '11:45:00', value: '11:45:00' },
-    { label: '12:00:00', value: '12:00:00' },
-    { label: '12:15:00', value: '12:15:00' },
-    { label: '12:30:00', value: '12:30:00' },
-    { label: '12:45:00', value: '12:45:00' },
-    { label: '13:00:00', value: '13:00:00' },
-    { label: '13:15:00', value: '13:15:00' },
-    { label: '13:30:00', value: '13:30:00' },
-    { label: '13:45:00', value: '13:45:00' },
-    { label: '14:00:00', value: '14:00:00' },
-    { label: '14:15:00', value: '14:15:00' },
-    { label: '14:30:00', value: '14:30:00' },
-    { label: '14:45:00', value: '14:45:00' },
-    { label: '15:00:00', value: '15:00:00' },
-    { label: '15:15:00', value: '15:15:00' },
-    { label: '15:30:00', value: '15:30:00' },
-    { label: '15:45:00', value: '15:45:00' },
-    { label: '16:00:00', value: '16:00:00' },
-    { label: '16:15:00', value: '16:15:00' },
-    { label: '16:30:00', value: '16:30:00' },
-    { label: '16:45:00', value: '16:45:00' },
-    { label: '17:00:00', value: '17:00:00' },
-    { label: '17:15:00', value: '17:15:00' },
-    { label: '17:30:00', value: '17:30:00' },
-    { label: '17:45:00', value: '17:45:00' },
-    { label: '18:00:00', value: '18:00:00' },
-    { label: '18:15:00', value: '18:15:00' },
-    { label: '18:30:00', value: '18:30:00' },
-    { label: '18:45:00', value: '18:45:00' },
-    { label: '19:00:00', value: '19:00:00' },
-    { label: '19:15:00', value: '19:15:00' },
-    { label: '19:30:00', value: '19:30:00' },
-    { label: '19:45:00', value: '19:45:00' },
-    { label: '20:00:00', value: '20:00:00' },
-    { label: '20:15:00', value: '20:15:00' },
-    { label: '20:30:00', value: '20:30:00' },
-    { label: '20:45:00', value: '20:45:00' },
-    { label: '21:00:00', value: '21:00:00' },
-    { label: '21:15:00', value: '21:15:00' },
-    { label: '21:30:00', value: '21:30:00' },
-    { label: '21:45:00', value: '21:45:00' },
-    { label: '22:00:00', value: '22:00:00' },
-    { label: '22:15:00', value: '22:15:00' },
-    { label: '22:30:00', value: '22:30:00' },
-    { label: '22:45:00', value: '22:45:00' },
-    { label: '23:00:00', value: '23:00:00' },
-    { label: '23:15:00', value: '23:15:00' },
-    { label: '23:30:00', value: '23:30:00' },
-    { label: '23:45:00', value: '23:45:00' },
+    { label: '00:00:00', value: '00:00' },
```

Comment:
Can't we generate these algorithmically to avoid a big static list?

Reply:
Hi @wblock , thanks for the suggestion. This is a const variable defined in the task service component. As doing this, it will avoid extra calculations to generate the options each time been used.